### PR TITLE
Check for Webpage name duplication. Returns error if name is a duplic…

### LINKED ISSUE #2300: Web page duplicate labels does not raise error properly, 500s
```
File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 112, in get_response
  response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/usr/local/lib/python2.7/site-packages/django/views/generic/base.py", line 69, in view
  return self.dispatch(request, *args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/django/utils/decorators.py", line 29, in _wrapper
  return bound_func(*args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 22, in _wrapped_view
  return view_func(request, *args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/django/utils/decorators.py", line 25, in bound_func
  return func(self, *args2, **kwargs2)
File "/app/codalab/apps/web/views.py", line 144, in dispatch
  return super(LoginRequiredMixin, self).dispatch(*args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/django/views/generic/base.py", line 87, in dispatch
  return handler(request, *args, **kwargs)
File "/app/codalab/apps/web/views.py", line 383, in post
  return super(CompetitionEdit, self).post(request, *args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/extra_views/advanced.py", line 150, in post
  return super(BaseUpdateWithInlinesView, self).post(request, *args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/extra_views/advanced.py", line 104, in post
  return self.forms_valid(form, inlines)
File "/app/codalab/apps/web/views.py", line 288, in forms_valid
  save_result = super(CompetitionEdit, self).forms_valid(form, inlines)
File "/usr/local/lib/python2.7/site-packages/extra_views/advanced.py", line 51, in forms_valid
  formset.save()
File "/usr/local/lib/python2.7/site-packages/django/forms/models.py", line 618, in save
  return self.save_existing_objects(commit) + self.save_new_objects(commit)
File "/usr/local/lib/python2.7/site-packages/django/forms/models.py", line 747, in save_new_objects
  self.new_objects.append(self.save_new(form, commit=commit))
File "/usr/local/lib/python2.7/site-packages/django/forms/models.py", line 877, in save_new
  obj.save()
File "/app/codalab/apps/web/models.py", line 628, in save
  return super(Page, self).save(*args, **kwargs)
File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 545, in save
  force_update=force_update, update_fields=update_fields)
File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 573, in save_base
  updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 654, in _save_table
  result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
File "/usr/local/lib/python2.7/site-packages/django/db/models/base.py", line 687, in _do_insert
  using=using, raw=raw)
File "/usr/local/lib/python2.7/site-packages/django/db/models/manager.py", line 232, in _insert
  return insert_query(self.model, objs, fields, **kwargs)
File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 1514, in insert_query
  return query.get_compiler(using=using).execute_sql(return_id)
File "/usr/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 903, in execute_sql
  cursor.execute(sql, params)
File "/usr/local/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
  return self.cursor.execute(sql, params)
File "/usr/local/lib/python2.7/site-packages/django/db/utils.py", line 99, in __exit__
  six.reraise(dj_exc_type, dj_exc_value, traceback)
File "/usr/local/lib/python2.7/site-packages/django/db/backends/util.py", line 53, in execute
  return self.cursor.execute(sql, params)

IntegrityError: duplicate key value violates unique constraint “idx_22330_web_page_label_5235asdf_uniq” DETAIL: Key (label, category_id, container_id)=(Evaluation, 1, 123) already exists.
```

### LINKED ISSUE #2332: Team Name Error Checking
```IntegrityError: duplicate key value violates unique constraint “teams_team_name_3146a213afbaca4e_uniq” DETAIL: Key (name, competition_id)=(Authors, 19527) already exists.```

Add a validator check to the team creation form to prevent this

### PR DESCRIPTION
**Pull request by BrendanMyers88**
Resolves #2300 

The purpose of this PR is to check if a Web page name is a duplicate of a Web page name already used in the competition. If the name is a duplicate, it will return the user to the form with a field-errror.

### CODE REVIEW

File: codalab/apps/web/forms.py [hunk 1]
```diff
@@ -140,6 +141,24 @@ class Meta:
...
+        cleaned_data = super(PageForm, self).clean()
+        label = cleaned_data.get('label')
+        if label:
+            existing_website = models.Page.objects.filter(
+                competition=self.instance.competition,
+                label=label
+            ).exclude(pk=self.instance.pk)
+            if existing_website.exists():
+                raise forms.ValidationError(
+                    {
```
Comment (ckcollab) [inline codalab/apps/web/forms.py]: I did this, looking at it now it's way too bulky, can we make this code a bit more compact?

File: codalab/apps/web/forms.py [hunk 2]
```diff
@@ -140,6 +141,24 @@ class Meta:
...
+                        'name': ['Invalid value for website name. A website already exists with that name.']
+                    },
+                    code='invalid'
```
Comment (ckcollab) [inline codalab/apps/web/forms.py]: can we figure out what this should actually be?

Comment (BrendanMyers88) [reply codalab/apps/web/forms.py]: code='invalid' is proper Validation Handling according to Django. 
  
https://docs.djangoproject.com/en/dev/ref/forms/validation/#raising-validationerror

File: codalab/apps/web/forms.py (hunk 1 above)
Comment (BrendanMyers88) [reply codalab/apps/web/forms.py]: Code is now on one line

**Comment by ckcollab** [comment]
:shipit: thanks for explaining what that new `code=` kwarg is!

**Pull request merged**